### PR TITLE
Wrote the multistats API call to return stats from multiple things

### LINKED ISSUE #4: Write a `multistats` API call
Takes a model_id and emissions scenario, searches for unique_ids and delegates to the stats() API call.

### PR DESCRIPTION
**Pull request by jameshiebert**
Gives the ability to request stats for multiple models and scenarios in a single request

Closes #4

### CODE REVIEW

File: ce/api/stats.py [hunk 1]
```diff
@@ -50,11 +55,18 @@ def stats(sesh, id_, time, area, variable):
                 }
             }
 
-    Raises:
-        Exception in several cases:
+        There are two cases semi-error cases which should be mentioned,
```
Comment (basilveerman) [inline ce/api/stats.py]: Words.

File: ce/api/util.py [hunk 2]
```diff
@@ -5,6 +5,7 @@
 import numpy.ma as ma
 from netCDF4 import Dataset
 from shapely.wkt import loads
+from modelmeta import *
```
Comment (basilveerman) [inline ce/api/util.py]: Normally this is what I do... and then you fix my code with named imports :wink:

File: ce/api/util.py [hunk 3]
```diff
@@ -80,3 +81,19 @@ def mean_datetime(datetimes):
     timestamps = [ dt.replace(tzinfo=timezone.utc).timestamp() for dt in datetimes ]
     mean = np.mean(timestamps)
     return datetime.fromtimestamp(mean, tz=timezone.utc)
+
+def search_for_unique_ids(sesh, ensemble_name='ce', model='', emission='', variable='', time=0):
+    query = sesh.query(DataFile.unique_id)\
+                  .join(DataFileVariable, EnsembleDataFileVariables, Ensemble, Run, Model, Emission)\
+                  .filter(Ensemble.name == ensemble_name)\
+                  .filter(DataFileVariable.netcdf_variable_name == variable)\
+                  .filter(Time.time_idx == time)\
```
Comment (basilveerman) [inline ce/api/util.py]: What's the purpose of filtering on time_idx?

File: ce/api/util.py [hunk 4]
```diff
@@ -80,3 +81,19 @@ def mean_datetime(datetimes):
     timestamps = [ dt.replace(tzinfo=timezone.utc).timestamp() for dt in datetimes ]
     mean = np.mean(timestamps)
     return datetime.fromtimestamp(mean, tz=timezone.utc)
+
+def search_for_unique_ids(sesh, ensemble_name='ce', model='', emission='', variable='', time=0):
+    query = sesh.query(DataFile.unique_id)\
+                  .join(DataFileVariable, EnsembleDataFileVariables, Ensemble, Run, Model, Emission)\
+                  .filter(Ensemble.name == ensemble_name)\
+                  .filter(DataFileVariable.netcdf_variable_name == variable)\
```
Comment (basilveerman) [inline ce/api/util.py]: Why is `variable` treated any differently than `model`/`emission`?

File: ce/api/util.py [hunk 5]
```diff
@@ -80,3 +81,19 @@ def mean_datetime(datetimes):
     timestamps = [ dt.replace(tzinfo=timezone.utc).timestamp() for dt in datetimes ]
     mean = np.mean(timestamps)
     return datetime.fromtimestamp(mean, tz=timezone.utc)
+
+def search_for_unique_ids(sesh, ensemble_name='ce', model='', emission='', variable='', time=0):
```
Comment (basilveerman) [inline ce/api/util.py]: Good to split this out.

File: ce/api/util.py (hunk 2 above)
Comment (jameshiebert) [inline ce/api/util.py]: Yeah, I don't *normally* like to do this, but we traverse across 7 ORM classes in the search, and that felt too verbose for me.

File: ce/api/util.py (hunk 3 above)
Comment (jameshiebert) [inline ce/api/util.py]: If you're searching for seasonal averages (e.g. pseudo-timestep 14), but there exist files that only have monthly averages, (e.g. 0-11), those files should not be included in the search results.

File: ce/api/util.py (hunk 4 above)
Comment (jameshiebert) [inline ce/api/util.py]: A single file can potentially contain multiple variables (though it's not recommended), but only ever represents a single model and a single emission scenario.

**Pull request merged**